packages/data/src/namespace-store/test/index.js

```diff
@@ -80,4 +85,18 @@ describe( 'controls', () => {
 
 		registry.select( 'store' ).getItems();
 	} );
+	it( 'returns undefined when action is dispatched unless it is a ' +
+		'promise', () => {
+		const actions = {
+			withPromise: () => new Promise( ( resolve ) => resolve( {} ) ),
+			normal: () => ( { type: 'NORMAL' } ),
+		};
+		registry.registerStore( 'store', {
+			reducer: () => {},
+			actions,
+		} );
+		expect( isPromise( registry.dispatch( 'store' ).withPromise() ) )
```

Comment:
Per https://github.com/WordPress/gutenberg/pull/14711#issuecomment-479994280, we should have an assertion for the expected _resolved_ value of the promise as well.

Reply:
ya good call.  I took this as incentive to expand the tests to cover most return scenarios (including ones similar to the use-case I have in a plugin I'm writing).  I think the test coverage better describes the expected behaviour now (which of course may still be up for discussion if desired).